### PR TITLE
Type Providers: fix Provided Entities caches dumping

### PR DESCRIPTION
**Pull request by DedSec256**
To improve the stability of the type providers tests, we should not use unstable provided entities (ProvidedType and ProvidedAssembly) identifiers. To determine the entity version, the FullName of the entity and the identifier of the type provider are sufficient.

### CODE REVIEW

**Comment by DedSec256** [comment]
I also need to check that in 212 everything is okay with this.

**Comment by DedSec256** [comment]
for 212 works, 
for TC too

File: rider-fsharp/testData/typeProviders/TypeProvidersCacheTest/checkCachesBeforeAndAfterReloading/gold/checkCachesBeforeAndAfterReloading.gold [hunk 1]
```diff
@@ -21,98 +21,98 @@ Type Providers:
...
-43 Microsoft.FSharp.Core.FSharpOption`1[FSharp.Text.RegexProvider.Regex,pattern="(?<Number>\\d+)"+MatchType] (from FSharp.Core, Version=4.7.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
-79 Microsoft.FSharp.Core.Unit (from FSharp.Core, Version=4.7.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
-84 ProviderImplementation.InternalType (from TestTypeProviders, Version=1.0.0.0, Culture=neutral, PublicKeyToken=333a98252ac829ae)
-83 ProviderImplementation.IPrintable (from TestTypeProviders, Version=1.0.0.0, Culture=neutral, PublicKeyToken=333a98252ac829ae)
-2 SimpleErasingProviderNamespace.AbstractType (from TestTypeProviders, Version=1.0.0.0, Culture=neutral, PublicKeyToken=333a98252ac829ae)
-3 SimpleErasingProviderNamespace.ClassWithInterfaces (from TestTypeProviders, Version=1.0.0.0, Culture=neutral, PublicKeyToken=333a98252ac829ae)
+3 FSharp.Configuration.AppSettings (from FSharp.Configuration, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null)
```
Comment (auduchinok) [inline rider-fsharp/testData/typeProviders/TypeProvidersCacheTest/checkCachesBeforeAndAfterReloading/gold/checkCachesBeforeAndAfterReloading.gold:24]: Does these ids need to be dumped? Is there some guarantee they won't change with the current approach?

**Comment by auduchinok** [comment]
Could you please target 212 before the merge and backport the commit afterwards, so the whole development occurs in the actual branch instead of stale-ish one?

**Pull request merged**